### ISSUE TITLE
Can't install Cilium’s kube-proxy replacement

### ISSUE THREAD

**Issue by handrea2009** [cilium/cilium#21334]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

I Can't install Cilium’s kube-proxy replacement on a k3s cluster.
The helm installation fails and shows the following error:

`INSTALLATION FAILED: template: cilium/templates/cilium-configmap.yaml:54:30: executing "cilium/templates/cilium-configmap.yaml" at <.Values.ipam.mode>: can't evaluate field mode in type interface {}`

I am using the following setting for the Cilium helm installation, cause I need to preserve source IP:

`      - operator.replicas=2
      - tunnel=disabled
      - autoDirectNodeRoutes=true
      - kubeProxyReplacement=strict
      - loadBalancer.mode=dsr
      - k8sServiceHost=10.130.42.84
      - k8sServicePort=6443
      - operator.image.useDigest=false
      - image.useDigest=false
      - ipv4-native-routing-cidr=10.0.0.0/16
      - enable-ipv4-masquerade=true
      - ipam=cluster-pool
      - enable-ipv4=true`

### Cilium Version

1.12.2 and 1.12.0

### Kernel Version

Linux rocky1 5.14.0-70.13.1.el9_0.x86_64 #1 SMP PREEMPT Wed May 25 21:01:57 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

# kubectl version
Client Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6+k3s1", GitCommit:"418c3fa858b69b12b9cefbcff0526f666a6236b9", GitTreeState:"clean", BuildDate:"2022-04-28T22:16:18Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"23", GitVersion:"v1.23.6+k3s1", GitCommit:"418c3fa858b69b12b9cefbcff0526f666a6236b9", GitTreeState:"clean", BuildDate:"2022-04-28T22:16:18Z", GoVersion:"go1.17.5", Compiler:"gc", Platform:"linux/amd64"}

### Relevant log output

```shell
INSTALLATION FAILED: template: cilium/templates/cilium-configmap.yaml:54:30: executing "cilium/templates/cilium-configmap.yaml" at <.Values.ipam.mode>: can't evaluate field mode in type interface {}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by handrea2009** [comment]
I have been able to complete the installation using the following helm setting:

`      - operator.replicas=2
      - tunnel=disabled
      - autoDirectNodeRoutes=true
      - kubeProxyReplacement=strict
      - loadBalancer.mode=dsr
      - k8sServiceHost={{ k3s_vars.apiserver_ip }}
      - k8sServicePort=6443
      - operator.image.useDigest=false
      - image.useDigest=false
      - enable-ipv4-masquerade=true
      - enable-ipv4=true
      - ipv4NativeRoutingCIDR="10.0.0.0/16"
      - ipam.mode=cluster-pool
`
So in place of `ipv4-native-routing-cidr` I have used `ipv4NativeRoutingCIDR`, and in place of `ipam` I have used `ipam.mode`

Thanks @aanm for helping on that

**Comment by vincentmli** [comment]
FYI: PR https://github.com/cilium/cilium/pull/21195 is to document the missing helm options

**Comment by pchaigno** [comment]
@vincentmli Should we also add `Fixes: #21334` to #21195?

**Comment by ieth0** [comment]
As the fix in #21195  doesn't include any feature, is there any chance it could be backported to the next release of 1.12?
